pkg/controller/v1beta1/inferenceservice/reconcilers/deployment/deployment_reconciler.go

```diff
@@ -148,7 +168,47 @@ func createRawDefaultDeployment(componentMeta metav1.ObjectMeta,
 		deployment.Spec.Strategy = *componentExt.DeploymentStrategy
 	}
 	setDefaultDeploymentSpec(&deployment.Spec)
-	return deployment
+
+	var isvcname string
```

Comment:
It would be better to separate the logic to add auth-proxy in a function. from this method, it only check if there is label about `auth-enabled` and it will call the new function related to oauth-proxy manifest. Then return oauth-proxy and error. and add it to deployment if there is no errors.
With it, you can directly test the oauth-proxy function as well.